src/include/daos_metrics.h

```diff
@@ -0,0 +1,268 @@
+	DM_IO_1M_2M,
+	DM_IO_2M_4M,
+	DM_IO_4M_INF,
+	DM_IO_BYSIZE_COUNT,
+};
+
+/** Distribution of i/o rpc calls based on size */
+typedef struct {
+	unsigned long	idp_total_call_cnt;
+	unsigned long	idp_buckets[DM_IO_BYSIZE_COUNT];
+} daos_iocall_dist_byprot_t;
+
+/** Distribution ids for update rpc calls based on protection/replication type */
+enum {
+	/** No Protection */
+	DM_IO_NO_PROT = 0,
+	/** Replication */
+	DM_IO_RP,
+	/** Erasure coding and partial update of a stripe */
+	DM_IO_EC_P,
+	/** Erasure coding and update of full stripe */
+	DM_IO_EC_F,
+	DM_IO_BYPROTO_COUNT,
+};
+
+/** Distribution of i/o rpc calls based on protection type */
+typedef struct {
+	unsigned long	ids_total_call_cnt;
+	unsigned long	ids_buckets[DM_IO_BYPROTO_COUNT];
+} daos_iocall_dist_bysize_t;
+
+/**
+ * Returns the daos metrics version in use by the daos library.
+ * The library is compatible if the major version number returned matches
+ * DAOS_METRICS_MAJOR_VERSION and minor version is greater than or equal to
+ * DAOS_METRICS_MINOR_VERSION.
+ *
+ * \param[out]	major	Major version number
+ * \param[out]	minor	Minor version number
+ *
+ * \return   		0 if metrics is enabled, 1 if metrics is disabled.
+ */
+int
+daos_metrics_get_version(int *major, int *minor);
+
+/**
+ * Returns the daos client rpc counters of the specified type. Refer enum daos_rpc_cntr_type
+ * for the types supported. 
+ *
+ * \param[in]		type	Type of the rpc counters requested. It can be a
+ * 				union of enums listed in daos_rpc_cntr_type.
+ * \param[in,out]	cntrs	Struct containing the counters. Only sub structures
+ * 				of the matching type will be populated. 
+ *
+ * \return   		0	 if metrics is enabled, 1 if metrics is disabled, -ve on failure.
+ */
+int
+daos_metrics_get_rpccntrs(int type, daos_all_rpc_cntrs_t *cntrs);
+
+/**
+ * Returns the daos client stats metric of the specified io_type. Refer enum daos_io_type for the
+ * io types supported. 
+ *
+ * \param[in]		io_type	I/O type for which the stats is requested. It can be a union of
+ * 				enums listed in daos_io_type.
+ * \param[in,out]	stats	Struct containing the stats. Only sub structures of the matching
+ * 				type will be populated. 
+ *
+ * \return   		0	 if metrics is enabled, 1 if metrics is disabled, -ve on failure.
+ */
+int
+daos_metrics_get_iostats(int io_type, daos_iostats_t *stats);
+
+/**
+ * Returns the daos client i/o rpc call count distribution based on the size of data
+ * transferred for the specified io_type. Refer enum daos_io_type for the io types supported.
+ *
+ * \param[in]		io_type	I/O type for which the distribution info is requested.
+ * 				It can be a union of enums listed in daos_io_type.
+ * \param[in,out]	dist	Struct containing the distribution info. Only sub structures
+ * 				of the matching type will be populated. 
+ *
+ * \return   		0	 if metrics is enabled, 1 if metrics is disabled, -ve on failure.
+ */
+int
+daos_metrics_get_io_distbysize(int io_type, daos_iocall_dist_bysize_t *dist);
```

Comment:
can we merge this function and daos_metrics_get_objupdate_distbyprot, into 1
and have a flag that says by_type or by_prot?
and we can callit daos_metrics_get_io_dist.